### PR TITLE
amp-bind: Improve e-commerce example

### PR DESCRIPTION
**Pull request by choumx**
In preparation for related code lab:

- Adds carousel slide indicator, color swatches, default var values, variable prices based on SKU, size and quantity selection and buttons, and inlines CSS.

/to @kmh287

### CODE REVIEW

File: examples/bind/ecommerce.amp.html [hunk 1]
```diff
@@ -5,59 +5,123 @@
...
+      width: 100%;
+    }
+    .hidden {
+      display: none;
     }
   </style>
 </head>
 
 <body>
-  <button onclick="AMP.toggleExperiment('amp-bind');window.location.href=window.location.href;">Toggle &lt;amp-bind&gt; experiment</button>
```
Comment (kmh287) [inline examples/bind/ecommerce.amp.html]: shouldn't we keep the experiment toggles in for now?

File: examples/bind/ecommerce.amp.html [hunk 2]
```diff
@@ -81,14 +145,18 @@
...
   -->
-  <amp-state id="shirts" [src]="'/bind/ecommerce/sizes?shirt=' + selectedIndex">
+  <amp-state id="shirts" [src]="'/bind/ecommerce/sizes?shirt=' + selected.id">
     <script type="application/json">
       {
         "0": {
           "color": "black",
-          "image": "./shirts/black.jpg"
+          "image": "./shirts/black.jpg",
+          "sizes": {
```
Comment (kmh287) [inline examples/bind/ecommerce.amp.html]: Maybe be a little more explicit in the comments above about why '0' has sizes but the others don't?

File: examples/bind/ecommerce.amp.html [hunk 3]
```diff
@@ -115,15 +183,139 @@
...
+    <h5 class="title">MEN'S COTTON CREW NECK<br>LONG SLEEVE T-SHIRT</h5>
+    <h6 class="rating"><span class="mdl-color-text--red">★★★★☆</span> <span class="mdl-color-text--grey">(14)</span></h6>
+
+    <amp-carousel type="slides" layout="fixed-height" height=250 on="slideChange:AMP.setState({selected: {slide: event.index}})">
+      <amp-img width=200 height=250 src="./shirts/black.jpg" [src]="shirts[selected.id].image"></amp-img>
+      <amp-img width=200 height=250 src="./shirts/black.jpg" [src]="shirts[selected.id].image"></amp-img>
+      <amp-img width=200 height=250 src="./shirts/black.jpg" [src]="shirts[selected.id].image"></amp-img>
+    </amp-carousel>
+
+    <p class="dots">
```
Comment (kmh287) [inline examples/bind/ecommerce.amp.html]: Can we make these clickable to move around the carousel?

File: examples/bind/ecommerce.amp.html [hunk 4]
```diff
@@ -115,15 +183,139 @@
...
+            <td><amp-img width=40 height=40 option=5 src="./shirts/swatch/light-gray.jpg"></amp-img></td>
+            <td><amp-img width=40 height=40 option=6 src="./shirts/swatch/navy.jpg"></amp-img></td>
+            <td><amp-img width=40 height=40 option=7 src="./shirts/swatch/wine.jpg"></amp-img></td>
+          </tr>
+        </table>
+      </amp-selector>
+    </div>
+
+    <div class="options price">
+      <h6>PRICE : <span [text]="shirts[selected.id].sizes[selectedSize] || '---'">---</span></h6>
```
Comment (kmh287) [inline examples/bind/ecommerce.amp.html]: Is this meant to be a heading?

File: examples/bind/ecommerce.amp.html [hunk 5]
```diff
@@ -115,15 +183,139 @@
...
+
+      <button class="mdl-button mdl-button--raised mdl-button--accent"
+          [class]="'mdl-button mdl-button--raised mdl-button--accent ' +
+            ((!selectedSize || !shirts[selected.id].sizes || shirts[selected.id].sizes[selectedSize]) ? '' : 'hidden')"
+          [disabled]="!selectedSize || !shirts[selected.id].sizes" disabled>
+        ADD TO BAG
+      </button>
+      <button class="mdl-button mdl-button--raised hidden"
+          [class]="'mdl-button mdl-button--raised ' +
+            ((!selectedSize || !shirts[selected.id].sizes || shirts[selected.id].sizes[selectedSize]) ? 'hidden' : '')">
```
Comment (kmh287) [inline examples/bind/ecommerce.amp.html]: Could you add a comment as to when this button is supposed to be shown vs hidden?

File: examples/bind/ecommerce.amp.html (hunk 1 above)
Comment (choumx) [inline examples/bind/ecommerce.amp.html]: Really there just for convenience, but `onclick` is not a valid AMP attribute. We can still use `AMP.toggleExperiment` in dev console.

File: examples/bind/ecommerce.amp.html (hunk 2 above)
Comment (choumx) [inline examples/bind/ecommerce.amp.html]: Good idea, done. I'll also spell this out more explicitly in the code lab.

File: examples/bind/ecommerce.amp.html (hunk 3 above)
Comment (choumx) [inline examples/bind/ecommerce.amp.html]: Good idea! Done.

File: examples/bind/ecommerce.amp.html (hunk 4 above)
Comment (choumx) [inline examples/bind/ecommerce.amp.html]: Subheading? 🤷‍♂️

File: examples/bind/ecommerce.amp.html (hunk 5 above)
Comment (choumx) [inline examples/bind/ecommerce.amp.html]: Sure, done.

File: examples/bind/ecommerce.amp.html [hunk 6]
```diff
@@ -146,6 +150,9 @@
       - Reintroduce variables in object keys (removed in #7582)
 
     <amp-state [src]="..." on="fetch:AMP.setState({"shirts": {selected: {id: event.response}}})"
+
+    Note that shirt with ID == "0" contains size data (but other IDs don't)
+    because it's the default shirt color at page load.
```
Comment (kmh287) [inline examples/bind/ecommerce.amp.html]: Optional: Add "Size data for the remaining indices will be downloaded via XHR when their corresponding options are selected." or something like that.

Comment (choumx) [inline examples/bind/ecommerce.amp.html]: Done.

**Pull request merged**